scipy/stats/_qmc.py

```diff
@@ -1022,12 +1022,12 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
     def _random(self, n: IntNumber = 1) -> np.ndarray:
         """Base LHS algorithm."""
         if self.centered:
-            samples: np.ndarray | float = 0.5
+            samples = np.array(0.5)
```

Comment:
Concerning to use 0d array to avoid an errant type complaint. Is this ever going to return a 0d array?

Reply:
I tested this and it does return an 0d array if you ask for 0 sample and 0 dim (also in the tests). I can revert if you prefer?

Reply:
The code
```python3
from scipy.stats.qmc import LatinHypercube
lhs = LatinHypercube(0, centered=True)
lhs.random(0)
```
does not return a 0d array; it returns a 2d array with shape (0, 0). I think that is the correct behavior. I was concerned with the possibility of this function returning a 0d array, which would not be correct.

Still, I am also not sure that we should change the way we write this code to satisfy mypy. Use of a Python float here is what you had in mind originally because it's more natural, and I think the rest of the SciPy codebase agrees.

Reply:
Sure, agreed. I just reverted this line.